embedding/csharp/Tizen.Flutter.Embedding.Tests/Channels/MethodChannelTests.cs

```diff
@@ -0,0 +1,203 @@
+                messenger.SendAsync(TEST_CHANNEL_NAME, Arg.Any<byte[]>())
+                         .Returns(codec.EncodeErrorEnvelope("E0001", "TEST_ERROR", "TEST_ERROR_DETAILS"));
+
+                var ex = await Assert.ThrowsAsync<FlutterException>(() =>
+                  {
+                      return channel.InvokeMethodAsync(call.Method, call.Arguments);
+                  });
+                Assert.Equal("E0001", ex.Code);
+                Assert.Equal("TEST_ERROR", ex.Message);
+                Assert.Equal("TEST_ERROR_DETAILS", ex.Details as string);
+            }
+
+            [Fact]
+            public void Requests_Null_Method_Name()
+            {
+                var messenger = Substitute.For<IBinaryMessenger>();
+                var channel = new MethodChannel(TEST_CHANNEL_NAME, StandardMethodCodec.Instance, messenger);
+
+                Assert.ThrowsAsync<ArgumentNullException>(() => channel.InvokeMethodAsync(null, null));
+            }
+
+            [Fact]
+            public void Requests_Null_Method_Call()
+            {
+                var messenger = Substitute.For<IBinaryMessenger>();
+                var channel = new MethodChannel(TEST_CHANNEL_NAME, StandardMethodCodec.Instance, messenger);
+
+                Assert.ThrowsAsync<ArgumentNullException>(() => channel.InvokeMethodAsync(null));
+            }
+        }
+
+        public class TheSetMethodCallHandlerMethod
+        {
+            [Fact]
+            public void Registers_Message_Handler()
+            {
+                var messenger = Substitute.For<IBinaryMessenger>();
+                var channel = new MethodChannel(TEST_CHANNEL_NAME, StandardMethodCodec.Instance, messenger);
+                channel.SetMethodCallHandler((call) =>
+                {
+                    return null;
+                });
+
+                messenger.Received().SetMessageHandler(Arg.Is<string>(x => x == TEST_CHANNEL_NAME),
+                                                       Arg.Any<BinaryMessageHandler>());
+            }
+
+            [Fact]
+            public void Unregisters_Message_Handler()
+            {
+                var messenger = Substitute.For<IBinaryMessenger>();
+                var channel = new MethodChannel(TEST_CHANNEL_NAME, StandardMethodCodec.Instance, messenger);
+                channel.SetMethodCallHandler(null);
+
+                messenger.Received().SetMessageHandler(Arg.Is<string>(x => x == TEST_CHANNEL_NAME), null);
+            }
+
+            [Fact]
+            public void Ensures_Error_Response_When_FlutterException_Is_Thrown()
+            {
+                var messenger = Substitute.For<IBinaryMessenger>();
+                var codec = StandardMethodCodec.Instance;
+                var channel = new MethodChannel(TEST_CHANNEL_NAME, StandardMethodCodec.Instance, messenger);
+                byte[] testCall = codec.EncodeMethodCall(new MethodCall("test", 1));
```

Comment:
Maybe
```suggestion
                byte[] testCall = codec.EncodeMethodCall(new MethodCall(TEST_METHOD_NAME, 1));
```
(also in line 188)